lib/platform/comment.ts

```diff
@@ -0,0 +1,57 @@
+import hasha from 'hasha';
+import { getCache } from '../util/cache/repository';
+import type {
+  EnsureCommentConfig,
+  EnsureCommentRemovalConfigByContent,
+  EnsureCommentRemovalConfigByTopic,
+} from './types';
+import { platform } from '.';
+
+export async function ensureComment(
+  commentConfig: EnsureCommentConfig
+): Promise<boolean> {
+  const { number, content } = commentConfig;
+  const topic = commentConfig.topic ?? '';
```

Comment:
For empty topic, it should be hash of content, I think

Reply:
I'm not sure if we've multiple comments without topic. 🤔

Reply:
Comment-based automerge

Reply:
Instead of using the raw topic as the key, what if you did this?
- If there's a topic, then the key is the topic hash, and the value is the content hash
- If there's no topic, then the key is the content hash, and the value is `null`

This way you could support multiple topic-less comments per PR in future if needed.

BTW I'm thinking about whether we need to "prune" this state, however if all we're storing is hashes then it's probably not a size concern.